### PR TITLE
Refactor SourceTest.java

### PR DESCRIPTION
**Pull request by Captain1653**
1. Thare are many lines look like that:
new ArrayList<String>
maybe it will be nice to replace it on new ArrayList<> (diamond operator).  [Documentation](https://docs.oracle.com/javase/7/docs/technotes/guides/language/type-inference-generic-instance-creation.html)
2. `try ... catch` in tests can be replaced on  [assertThrows](https://junit.org/junit4/javadoc/latest/org/junit/Assert.html#assertThrows(java.lang.Class,%20org.junit.function.ThrowingRunnable))

Kind regards

### CODE REVIEW

**Comment by octonato** [comment]
OK TO TEST

**Review by octonato — APPROVED** [review]
LGTM

Nice improvement.

File: akka-stream-tests/src/test/java/akka/stream/javadsl/SourceTest.java [hunk 1]
```diff
@@ -966,21 +951,26 @@ public void cycleSourceMustGenerateSameSequenceInRepeatedFashion() throws Except
...
-          // #cycle-error
-          .toCompletableFuture()
-          .get();
-    } catch (ExecutionException e) {
-      throw e.getCause();
-    }
+  @Test
+  public void cycleSourceMustThrow() {
+    ExecutionException exception =
+        Assert.assertThrows(
```
Comment (johanandren) [inline akka-stream-tests/src/test/java/akka/stream/javadsl/SourceTest.java:957]: Perhaps do a static import of this like with other static assert methods.

**Review by johanandren — APPROVED** [review]
LGTM but I left one, potential, tiny, improvement suggestion.

File: akka-stream-tests/src/test/java/akka/stream/javadsl/SourceTest.java (hunk 1 above)
Comment (Captain1653) [reply akka-stream-tests/src/test/java/akka/stream/javadsl/SourceTest.java:957]: It's the cool suggestion, but I couldn't. `JUnitSuiteLike` has own `assertThrows` (different signature). I had compilation exception even with static import. I should specified the full name.  Am I wrong?

Comment (johanandren) [reply akka-stream-tests/src/test/java/akka/stream/javadsl/SourceTest.java:957]: Ah, thanks for explaining, let's merge as is then :)

**Pull request merged**